url354ce1a2d1_dagaren_gladiator_chess_tgz-pJ8-es_dagaren_gladiator_main_MainJ8/src/es/dagaren/gladiator/representation/BitboardPosition.java

```diff
@@ -0,0 +1,2907 @@
+      castlingShortKingSquares[Colour.WHITE.index][1] = Square.g1;
+      
+      castlingShortKingSquares[Colour.BLACK.index] = new Square[2];
+      castlingShortKingSquares[Colour.BLACK.index][0] = Square.f8;
+      castlingShortKingSquares[Colour.BLACK.index][1] = Square.g8;
+      
+      
+      //Se inicializan los arrays de casillas relacionadas con las capturas al paso
+      for(Square  s = Square.a1; s != null; s = s.getNext())
+      {
+         if(s.getRank() == Rank._3)
+         {
+            enPassantSquareSource[s.index] = s.getNextInFile();
+            enPassantSquareDestination[s.index] = null;
+         }
+         else if(s.getRank() == Rank._6)
+         {
+            enPassantSquareSource[s.index] = s.getPreviousInFile();
+            enPassantSquareDestination[s.index] = null;
+         }
+         else if(s.getRank() == Rank._4)
+         {
+            enPassantSquareDestination[s.index] = s.getPreviousInFile();
+            enPassantSquareSource[s.index] = null;
+         }
+         else if(s.getRank() == Rank._5)
+         {
+            enPassantSquareDestination[s.index] = s.getNextInFile();
+            enPassantSquareSource[s.index] = null;
+         }
+         else
+         {
+            enPassantSquareSource[s.index] = null;
+            enPassantSquareDestination[s.index] = null;
+         }    
+      }
+   } // Fin de iniciarlizar()
+   
+   /**
+    * 
+    *
+    */
+   protected void updateOcupied()
+   {
+      //long ti = System.currentTimeMillis();
+      
+      //Se calcula el bitboard de ocupación de piezas blancas
+      bbWhiteOccupation = bbPiecesOccupation[Piece.WHITE_PAWN.index] |
+                     bbPiecesOccupation[Piece.WHITE_ROOK.index] |
+                     bbPiecesOccupation[Piece.WHITE_KNIGHT.index] |
+                     bbPiecesOccupation[Piece.WHITE_BISHOP.index] |
+                     bbPiecesOccupation[Piece.WHITE_QUEEN.index] |
+                     bbPiecesOccupation[Piece.WHITE_KING.index];
+         
+      //Se calcula el bitboard de ocupación de piezas negras
+      bbBlackOccupation =  bbPiecesOccupation[Piece.BLACK_PAWN.index] |
+                     bbPiecesOccupation[Piece.BLACK_ROOK.index] |
+                     bbPiecesOccupation[Piece.BLACK_KNIGHT.index] |
+                     bbPiecesOccupation[Piece.BLACK_BISHOP.index] |
+                     bbPiecesOccupation[Piece.BLACK_QUEEN.index] |
+                     bbPiecesOccupation[Piece.BLACK_KING.index];
+      
+      bbColourOccupation[Colour.WHITE.index] = bbWhiteOccupation;
+      bbColourOccupation[Colour.BLACK.index] = bbBlackOccupation;
+      
+      //Se calcula el bitboard de las casillas ocupadas
+      bbOccupation = bbWhiteOccupation | bbBlackOccupation;
+      
+      //long to = System.currentTimeMillis();
+      //System.out.println("Tiempo de actualizacion de ocupadas: "+(to-ti)+ " milisegundos");
+      
+   } //Fin de actualizarOcupadas()
+   
+
+   public void updateAttackedFromFull()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 66 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 1078-1298 into a separate method.